docs/reference/eql/eql.asciidoc

```diff
@@ -792,6 +792,44 @@ results by using <<get-async-eql-status-api,get async EQL status API>>.
 You can also manually delete saved synchronous searches using the
 <<delete-async-eql-search-api,delete async EQL search API>>.
 
+[discrete]
+[[eql-circuit-breaker]]
+=== EQL circuit breaker settings
+
+When a <<eql-sequences, sequence>> query is executed, the node handling the query,
+needs to keep some structures in memory which are needed by the algorithm which
+implements the sequence matching. When large amounts of data need to be processed,
+and/or a large amount of matched sequences is requested by the user (by setting the
+<<eql-search-api-params-size, size>> query param), the memory occupied by those
+structures could potentially exceed the available memory in the JVM. This would cause
+and `OutOfMemory` exception which would bring down the node.
+
+To prevent this from happening, a special <<circuit-breaker, circuit breaker>> is used,
+which limits the memory allocation during the execution of a <<eql-sequences, sequence>>
+query, so instead of an `OutOfMemory` exception, a `org.elasticsearch.common.breaker.CircuitBreakingException`
+is thrown, and a descriptive error message is returned to the user.
```

Comment:
If the circuit breaker doesn't do its job, it doesn't mean an OOM is thrown. How about the following message?
```suggestion
query. When the breaker is triggered, an `org.elasticsearch.common.breaker.CircuitBreakingException`
is thrown and a descriptive error message is returned to the user.
```

Reply:
I'm torn between `an org......` and `a ...CircuitBreakingException` :D

Reply:
Leaving out the package name or even Exception (which are developers facing implementation details) is preferred. Just an "error message" or similar is enough. Not all users are developers.